### PR TITLE
Abstract subsystems from state machine

### PR DESCRIPTION
**Pull request by AlenaNguyen**
my biggest regret in life is not making the LEDs blue and yellow for our team colors. it keeps me up at night.

In other news, the new state machine works and the code looks so pretty now :D

### CODE REVIEW

**Comment by alicekuznetsov** [comment]
I can't believe you've done this. The robot simply was not functional with teal and orange lights.

File: src/main/java/frc/robot/commands/Autos/ShootSequence.java [hunk 1]
```diff
@@ -65,17 +64,16 @@ public ShootSequence(StateMachine subStateMachine, Climber subClimber, Drivetrai
...
             Commands.waitSeconds(constShooter.AUTO_PREP_NONE_DELAY.in(Units.Seconds)),
 
             // Reset subsystems to chill
             Commands.deferredProxy(() -> subStateMachine
-                .tryState(RobotState.NONE, subStateMachine, subClimber, subDrivetrain, subElevator, subIntake,
-                    subLEDs, subTransfer,
-                    subShooter)))
+                .tryState(RobotState.NONE)),
+
+            Commands.runOnce(() -> subStateMachine.setTargetState(TargetState.PREP_VISION)))
```
Comment (TaylerUva) [inline src/main/java/frc/robot/commands/Autos/ShootSequence.java]: This line should be removed as it was from merge conflict.

**Review by TaylerUva — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/commands/Autos/ShootSequence.java (hunk 1 above)
Comment (TaylerUva) [inline src/main/java/frc/robot/commands/Autos/ShootSequence.java]:
```suggestion
```

**Review by TaylerUva — APPROVED** [review]

File: src/main/java/frc/robot/commands/Autos/ShootSequence.java [hunk 2]
```diff
@@ -65,17 +64,15 @@ public ShootSequence(StateMachine subStateMachine, Climber subClimber, Drivetrai
...
                 .until(() -> !subTransfer.getGamePieceStored())),
 
             Commands.waitSeconds(constShooter.AUTO_PREP_NONE_DELAY.in(Units.Seconds)),
 
             // Reset subsystems to chill
             Commands.deferredProxy(() -> subStateMachine
-                .tryState(RobotState.NONE, subStateMachine, subClimber, subDrivetrain, subElevator, subIntake,
-                    subLEDs, subTransfer,
-                    subShooter)))
+                .tryState(RobotState.NONE)),
```
Comment (TaylerUva) [inline src/main/java/frc/robot/commands/Autos/ShootSequence.java]:
```suggestion
                .tryState(RobotState.NONE))
```

**Review by alicekuznetsov — APPROVED** [review]
its beautiful

**Pull request merged**